org.jacoco.core/src/org/jacoco/core/internal/instr/InstrSupport.java

```diff
@@ -238,7 +236,7 @@ public static void push(final MethodVisitor mv, final int value) {
 	 */
 	public static ClassReader classReaderFor(final byte[] b) {
 		final byte[] originalVersion = new byte[] { b[4], b[5], b[6], b[7] };
-		if (getVersionMajor(b) == Opcodes.V12 + 1) {
+		if (b[6] == 0x00 && b[7] == 0x39) {
```

Comment:
Magic number? The intend was more clear with `Opcodes.V12 + 1`.

Reply:
@marchof inlining of former implementation of `getVersionMajor` into this method results in

```java
final int majorVersion = (short) (((b[MAJOR_VERSION_INDEX] & 0xFF) << 8)
		| (b[MAJOR_VERSION_INDEX + 1] & 0xFF));

if (majorVersion == Opcodes.V12 + 1)
```

which IMO contains far more magic and far less readable than

```
if (b[6] == 0x00 && b[7] == 0x39)
```

Can also add comment.

Readable and non-magical alternative

```
final int majorVersion = java.nio.ByteBuffer.wrap(b)
		.order(java.nio.ByteOrder.BIG_ENDIAN).getShort(6);
```

looks like a bit of overkill.

Reply:
BTW numbers `6`, `7` and `0x39` are even shown in https://en.wikipedia.org/wiki/Java_class_file 😉

Reply:
So why inlining? If we keep the previous method (in addition to the new Reader version) we have testable and readable code ;)